### PR TITLE
add darkmode toggle

### LINKED ISSUE #67: Dark mode selector
Add a dark mode switch in the corner or something. Hack Club's theme-ui theme already supports dark mode, so it would just be an issue of using [this](https://theme-ui.com/color-modes#setting-the-color-mode) to switch themes between `default` and `dark`.

### PR DESCRIPTION
**Pull request by bellesea**
Closes #67

### CODE REVIEW

File: frontend/components/ColorButton.tsx [hunk 1]
```diff
@@ -0,0 +1,23 @@
+import { Avatar, Box, Container, Flex, Heading, SxProp, Text } from "theme-ui";
+import { Moon } from 'react-feather'
+
+const ColorButton = ({...props }) => (
+    <Box
```
Comment (cjdenio) [inline frontend/components/ColorButton.tsx]: `IconButton` might make more sense here

**Review by cjdenio — CHANGES_REQUESTED** [review]
lgtm, just one tiny change

(also run `yarn format:write`) to fix CI

File: frontend/components/ColorButton.tsx (hunk 1 above)
Comment (bellesea) [reply frontend/components/ColorButton.tsx]: ok! will make changes tmr :))

Comment (bellesea) [reply frontend/components/ColorButton.tsx]: changed my mind and decided i don't need sleep, made the edits!

File: frontend/components/ColorButton.tsx [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+    as="button"
+    {...props}
+    sx={{
+      backgroundColor: "transparent",
+      border: 0,
+      color: "inherit",
+      cursor: "pointer",
+      position: "absolute",
+      left: 200, // this is not very responsive css, we need to fix it later lmao
+      top: 26,
```
Comment (anirudhb) [inline frontend/components/ColorButton.tsx]: Please fix this now instead of later

**Review by anirudhb — CHANGES_REQUESTED** [review]

File: frontend/components/ColorButton.tsx [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+    }}
+  >
+    <Moon size={26} />
```
Comment (cfanoulis) [inline frontend/components/ColorButton.tsx]: shouldn't this change according to mode?

**Review by cfanoulis — COMMENTED** [review]
a suggestion

File: frontend/components/ColorButton.tsx (hunk 3 above)
Comment (bellesea) [reply frontend/components/ColorButton.tsx]: @cfanoulis do you mean the size of the moon?

Comment (cfanoulis) [reply frontend/components/ColorButton.tsx]: No, the icon itself. When it's light mode, it's a moon and when its dark mode its a sun.

File: frontend/components/ColorButton.tsx [hunk 4]
```diff
@@ -0,0 +1,32 @@
+import { IconButton } from "theme-ui";
+import { Moon, Sun } from "react-feather";
+import { useColorMode } from "theme-ui";
+
+function changeToggleButton({ ...props }) {
+  const [colorMode] = useColorMode();
```
Comment (cfanoulis) [inline frontend/components/ColorButton.tsx]: this can just be a prop on the button, instead of using the hook.

**Review by cfanoulis — CHANGES_REQUESTED** [review]
feedback on icon changing handling

File: frontend/components/ColorButton.tsx [hunk 5]
```diff
@@ -0,0 +1,32 @@
+import { IconButton } from "theme-ui";
+import { Moon, Sun } from "react-feather";
+import { useColorMode } from "theme-ui";
+
+function changeToggleButton({ ...props }) {
```
Comment (anirudhb) [inline frontend/components/ColorButton.tsx]: `export default function ColorButton` and remove the export below

File: frontend/components/ColorButton.tsx [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+  const [colorMode] = useColorMode();
+
+  let icon;
+
+  if (colorMode === "dark") {
+    icon = <Sun />;
+  } else {
+    icon = <Moon />;
+  }
```
Comment (anirudhb) [inline frontend/components/ColorButton.tsx]:
```suggestion
  const icon = colorMode == "dark" ? <Sun /> : <Moon />;
```

File: frontend/layouts/dashboard.tsx [hunk 7]
```diff
@@ -87,6 +89,11 @@ function SidebarHeader({ avatar }: { avatar?: string }) {
     >
       <Avatar src={avatar} />
       <Box sx={{ flexGrow: 1 }} />
+      <ColorButton
+        onClick={(e) => {
+          setColorMode(colorMode === "default" ? "dark" : "default");
+        }}
+      />
```
Comment (anirudhb) [inline frontend/layouts/dashboard.tsx]: Maybe refactor this into the component itself and rename to `ColorSwitcher`

**Review by anirudhb — CHANGES_REQUESTED** [review]
nits

File: frontend/layouts/dashboard.tsx [hunk 8]
```diff
@@ -5,8 +5,9 @@ import { PropsWithChildren, useEffect } from "react";
 import useSWR from "swr";
 import { Avatar, Box, Container, Flex, Heading, SxProp, Text } from "theme-ui";
 import fetchApi from "../lib/fetch";
-
+import { useColorMode } from "theme-ui";
```
Comment (anirudhb) [inline frontend/layouts/dashboard.tsx]: Remove this line

**Review by anirudhb — CHANGES_REQUESTED** [review]
lgtm! thanks for sticking with us Belle

**Comment by bellesea** [comment]
> lgtm! thanks for sticking with us Belle

thanks for helping me!

**Review by anirudhb — APPROVED** [review]

**Pull request merged**